src/HonzaBotner.Discord/Command/Permissions/Entities/PermissionEntry.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace HonzaBotner.Discord.Command.Permissions.Entities
+{
+    public class PermissionEntry
+    {
+        public Guid Id { get; set; }
+
+        [Required] public PermissionEntryType Type { get; set; }
+
+        [Required] public PermissionEntryTarget Target { get; set; }
+
+        [Required] public ulong TargetId { get; set; }
+
+        [Required] public string Permission { get; set; }
```

Comment:
Look at these warnings. If these fields are really required, you can suppress this warning with this little hack, since EF doesn't support nullable types yet.
```suggestion
        [Required] public string Permission { get; set; } = null!
```